src/top.rs

```diff
@@ -358,10 +396,24 @@ impl Link {
                         dst: sent.dst,
                         message: sent.protocol,
                     };
+
+                    // If the communication is localhost, use a real ip address
+                    // of the host, instead of a localhost address to deliver a
+                    // message. One of them is guaranteed not to be a loopback
+                    // address.
+                    let ip_addr = if sent.src.ip().is_loopback() {
```

Comment:
We could try to workaround this and make communication work differently, however I don't know whether this is a good idea.
Currently localhost communication depends on e.g `192.168.0.1 <-> 127.0.0.1`. If we made it e.g. `127.0.0.1 <-> 127.0.0.1` then topology wouldn't be able to determine to which link it belongs.

One idea would be to make world `send_message`, host-aware, but not sure about that.

Reply:
This seems reasonable to me.